networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlockAndBlobsSidecarByRootMessageHandler.java

```diff
@@ -0,0 +1,163 @@
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.metrics.TekuMetricCategory;
+import tech.pegasys.teku.infrastructure.ssz.primitive.SszBytes32;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.networking.eth2.peers.Eth2Peer;
+import tech.pegasys.teku.networking.eth2.rpc.core.PeerRequiredLocalMessageHandler;
+import tech.pegasys.teku.networking.eth2.rpc.core.ResponseCallback;
+import tech.pegasys.teku.networking.eth2.rpc.core.RpcException;
+import tech.pegasys.teku.networking.eth2.rpc.core.RpcException.ResourceUnavailableException;
+import tech.pegasys.teku.networking.p2p.rpc.StreamClosedException;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.versions.eip4844.SignedBeaconBlockAndBlobsSidecar;
+import tech.pegasys.teku.spec.datastructures.networking.libp2p.rpc.BeaconBlockAndBlobsSidecarByRootRequestMessage;
+import tech.pegasys.teku.storage.client.RecentChainData;
+
+public class BeaconBlockAndBlobsSidecarByRootMessageHandler
+    extends PeerRequiredLocalMessageHandler<
+        BeaconBlockAndBlobsSidecarByRootRequestMessage, SignedBeaconBlockAndBlobsSidecar> {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Spec spec;
+  private final UInt64 eip4844ForkEpoch;
+  private final RecentChainData recentChainData;
+  private final Counter totalBlockAndBlobsSidecarRequestedCounter;
+  private final LabelledMetric<Counter> requestCounter;
+
+  public BeaconBlockAndBlobsSidecarByRootMessageHandler(
+      final Spec spec,
+      final UInt64 eip4844ForkEpoch,
+      final MetricsSystem metricsSystem,
+      final RecentChainData recentChainData) {
+    this.spec = spec;
+    this.eip4844ForkEpoch = eip4844ForkEpoch;
+    this.recentChainData = recentChainData;
+    requestCounter =
+        metricsSystem.createLabelledCounter(
+            TekuMetricCategory.NETWORK,
+            "rpc_block_and_blobs_sidecar_by_root_requests_total",
+            "Total number of block and blobs sidecar by root requests received",
+            "status");
+    totalBlockAndBlobsSidecarRequestedCounter =
+        metricsSystem.createCounter(
+            TekuMetricCategory.NETWORK,
+            "rpc_block_and_blobs_sidecar_by_root_requested_total",
+            "Total number of block and blobs sidecar requested in accepted block and blobs sidecar by root requests from peers");
+  }
+
+  @Override
+  public Optional<RpcException> validateRequest(
+      final String protocolId, final BeaconBlockAndBlobsSidecarByRootRequestMessage request) {
+
+    final UInt64 finalizedEpoch = recentChainData.getFinalizedEpoch();
+    final UInt64 currentEpoch =
+        recentChainData
+            .getCurrentEpoch()
+            .orElse(UInt64.ZERO)
+            .minusMinZero(MIN_EPOCHS_FOR_BLOBS_SIDECARS_REQUESTS);
+    final UInt64 minimumRequestEpoch = finalizedEpoch.max(currentEpoch).max(eip4844ForkEpoch);
```

Comment:
Given the current epoch has to be greater than the finalized epoch doesn't this boil down to just:
```suggestion
    final UInt64 minimumRequestEpoch = currentEpoch.max(eip4844ForkEpoch);
```

Reply:
I named it bad, it is not the current epoch but `current_epoch - MIN_EPOCHS_FOR_BLOBS_SIDECARS_REQUESTS`. Renamed it to minEpochForBlobsSidecar